providers/databricks/src/airflow/providers/databricks/utils/openlineage.py

```diff
@@ -0,0 +1,328 @@
+    """
+    from airflow.providers.openlineage.plugins.adapter import OpenLineageAdapter
+
+    # Generate same OL run id as is generated for current task instance
+    return OpenLineageAdapter.build_task_instance_run_id(
+        dag_id=task_instance.dag_id,
+        task_id=task_instance.task_id,
+        logical_date=_get_logical_date(task_instance),
+        try_number=task_instance.try_number,
+        map_index=task_instance.map_index,
+    )
+
+
+# todo: move this run_id logic into OpenLineage's listener to avoid differences
+def _get_ol_dag_run_id(task_instance) -> str:
+    from airflow.providers.openlineage.plugins.adapter import OpenLineageAdapter
+
+    return OpenLineageAdapter.build_dag_run_id(
+        dag_id=task_instance.dag_id,
+        logical_date=_get_logical_date(task_instance),
+        clear_number=_get_dag_run_clear_number(task_instance),
+    )
+
+
+def _get_parent_run_facet(task_instance):
+    """
+    Retrieve the ParentRunFacet associated with a specific Airflow task instance.
+
+    This facet helps link OpenLineage events of child jobs - such as queries executed within
+    external systems (e.g., Databricks) by the Airflow task - to the original Airflow task execution.
+    Establishing this connection enables better lineage tracking and observability.
+    """
+    from openlineage.client.facet_v2 import parent_run
+
+    from airflow.providers.openlineage.conf import namespace
+
+    parent_run_id = _get_ol_run_id(task_instance)
+    root_parent_run_id = _get_ol_dag_run_id(task_instance)
+
+    return parent_run.ParentRunFacet(
+        run=parent_run.Run(runId=parent_run_id),
+        job=parent_run.Job(
+            namespace=namespace(),
+            name=f"{task_instance.dag_id}.{task_instance.task_id}",
+        ),
+        root=parent_run.Root(
+            run=parent_run.RootRun(runId=root_parent_run_id),
+            job=parent_run.RootJob(
+                name=task_instance.dag_id,
+                namespace=namespace(),
+            ),
+        ),
+    )
+
+
+def _run_api_call(hook: DatabricksSqlHook, query_ids: list[str]) -> list[dict]:
+    """Retrieve execution details for specific queries from Databricks's query history API."""
+    if not hook._token:
+        # This has logic for token initialization
+        hook.get_conn()
+
+    # https://docs.databricks.com/api/azure/workspace/queryhistory/list
+    response = requests.get(
+        url=f"https://{hook.host}/api/2.0/sql/history/queries",
+        headers={"Authorization": f"Bearer {hook._token}"},
+        data=json.dumps({"filter_by": {"statement_ids": query_ids}}),
+    )
+    if response.status_code != 200:
+        log.warning(
+            "OpenLineage could not retrieve Databricks queries details. API error received: `%s`: `%s`",
+            response.status_code,
+            response.text,
+        )
+        return []
+
+    return response.json()["res"]
+
+
+def _get_queries_details_from_databricks(
```

Comment:
How much does the `DatabricksSqlHook` performance degrade by adding fetching this additional metadata from Databricks?